vspace/src/main/webapp/WEB-INF/views/staff/modules/module.jsp

```diff
@@ -7,15 +7,50 @@
 
 <script>
 //# sourceURL=click.js
-$(document).ready(function($) {
+var slideIdToDelete = 0;
+//------------Deleting Slides-------------------
+function deleteSlide(slideId, hasSequence) {
+	if(hasSequence == 0) {
+		$.ajax({
+	        url: "<c:url value="/staff/module/${module.id}/slide/" />" + slideId +'?${_csrf.parameterName}=${_csrf.token}',
+	        type: 'DELETE',
+	        cache       : false,
+	        contentType : false,
+	        success: function(data) {
+	        	$("#"+slideId).closest('.slide').remove();
+	        },
+	        error: function(data) {
+	        	var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+	        	$('.error').append(alert);
+	        }
+	    });
+	} else {
+		slideIdToDelete = slideId;
+		$("#deleteSlideAlert").show();
```

Comment:
let's make it an ajax call to check if a slide is used in a sequence, otherwise the info might very out of date (e.g. someone works on something else on the page or takes a break and then comes back). Someone else might have added the slide to sequence or removed it in the meantime. To reduce the risk, you should check right before you're about to delete the slide and then show a warning.

Reply:
Okay